scripts/app.js

```diff
@@ -0,0 +1,356 @@
+      }
+    });
+  });
+}
+
+async function getGitCommits() {
+  return new Promise((resolve, reject) => {
+    exec("git cherry", function(error, stdout, stderr) {
+      if (error !== null) {
+        console.log("exec error: " + error);
+        reject(error);
+      } else {
+        const cleanedup = stdout.replace(/\+ /g, "");
+        let commitarray = cleanedup.split(/\r\n|\r|\n/);
+        commitarray.length = commitarray.length - 1; //remove the last, dummy element that results from splitting on newline
+        if (commitarray.length == 1) {
+          const firstCommit = commitarray[0];
+          const lastCommit = null;
+          resolve({ firstCommit, lastCommit });
+        } else {
+          const firstCommit = commitarray[0];
+          const lastCommit = commitarray[commitarray.length - 1];
+          resolve({ firstCommit, lastCommit });
+        }
+      }
+    });
+  });
+}
+async function getGitPatchFromLocal(){
+  return new Promise((resolve, reject) => {
+    exec("git diff > myPatch.patch", function(error, stdout, stderr) {
+      if (error !== null) {
+        console.log("error generating patch: ", error);
+        reject(error);
+      } else {
+        fs.readFile("myPatch.patch", "utf8", function(err, data) {
+          if (err) {
+            console.log("error reading patch file: ", err);
+            reject(err);
+          }
+          resolve(data);
+        });
+      }
+    });
+  })
+}
+async function getGitPatchFromCommits(firstCommit, lastCommit) {
+  //need to delete patch file?
+  return new Promise((resolve, reject) => {
+    if (lastCommit === null) {
+      let patchCommand = "git show HEAD > myPatch.patch";
+      exec(patchCommand, function(error, stdout, stderr) {
+        if (error !== null) {
+          console.log("error generating patch: ", error);
+          reject(error);
+        } else {
+          fs.readFile("myPatch.patch", "utf8", function(err, data) {
+            if (err) {
+              console.log("error reading patch file", err);
+              reject(err);
+            }
+            resolve(data);
+          });
+        }
+      });
+    } else {
+      let patchCommand =
+        "git diff " + firstCommit + "^..." + lastCommit + " > myPatch.patch";
+      exec(patchCommand, function(error, stdout, stderr) {
+        if (error !== null) {
+          console.log("error generating patch: ", error);
+          reject(error);
+        } else {
+          fs.readFile("myPatch.patch", "utf8", function(err, data) {
+            if (err) {
+              console.log("error reading patch file ", err);
+              reject(err);
+            }
+            resolve(data);
+          });
+        }
+      });
+    }
+  });
+}
+
+async function main() {
+  const buildSize = process.argv[2];
+  const patchFlag = process.argv[3];
+
+
+  let missingFlag = false
+  if(buildSize === undefined){
+    console.log('You need a build size flag("repo" or "world") in your make command')
```

Comment:
The interface for this needs a little more care. We will not require "repo" or "world" because the writer might be building this commit and nothing more. But if they do provide 2 args to this script -- we should validate that the arguments match the values expected. It's fine to make assumptions but ordering of parameters but we should make sure they haven't switched them. Also it will just be easier to assume the optional parameter ("repo" or "world") is last.